### PR TITLE
refactor(select): rewrite with hook

### PR DESCRIPTION
**Pull request by hengkx**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   rewrite with hook.        |
| 🇨🇳 Chinese |    使用hook重写。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/auto-complete/index.tsx [hunk 1]
```diff
@@ -37,13 +34,13 @@ function isSelectOptionOrSelectOptGroup(child: any): Boolean {
   return child && child.type && (child.type.isSelectOption || child.type.isSelectOptGroup);
 }
 
-const AutoComplete: React.ForwardRefRenderFunction<Select, AutoCompleteProps> = (props, ref) => {
+const AutoComplete: React.ForwardRefRenderFunction<unknown, AutoCompleteProps> = (props, ref) => {
```
Comment (zombieJ) [inline components/auto-complete/index.tsx:37]: 不能改 unkown，这样 ref 出来里面的方法就丢了。

Comment (hengkx) [reply components/auto-complete/index.tsx:37]: 那就只能是rcselect 了 ref并不是select

Comment (zombieJ) [reply components/auto-complete/index.tsx:37]: 可以 as，只要 focus 和 blur 这两个有了就行了

Comment (hengkx) [reply components/auto-complete/index.tsx:37]: select 是 函数 组件 里面 没有 那两个方法啊🤣🤣🤣

Comment (hengkx) [reply components/auto-complete/index.tsx]: 只要这两个方法 我改成 RCSelect<any> 吧

File: components/select/index.tsx [hunk 2]
```diff
@@ -35,148 +35,127 @@ export interface SelectProps<VT>
...
-    let mergedNotFound: React.ReactNode;
-    if (notFoundContent !== undefined) {
-      mergedNotFound = notFoundContent;
-    } else if (mode === 'combobox') {
-      mergedNotFound = null;
-    } else {
-      mergedNotFound = renderEmpty('Select');
-    }
+  const prefixCls = getPrefixCls('select', customizePrefixCls);
+  const mode = getMode();
```
Comment (zombieJ) [inline components/select/index.tsx:81]: 看起来 getMode 方法用不到了，直接 `useMemo` 吧

Comment (hengkx) [reply components/select/index.tsx]: 脑袋疼 命名 难

**Comment by afc163** [comment]
#27474

**Pull request merged**